include/libtorrent/kademlia/get_item.hpp

```diff
@@ -43,7 +43,7 @@ namespace libtorrent { namespace dht
 class get_item : public find_data
 {
 public:
-	typedef boost::function<bool(item&, bool)> data_callback;
+	typedef boost::function<void(item&, bool)> data_callback;
```

Comment:
oh, Then I have different understanding for nodes_callback. I think it's like a results not for traversal, but for put(just like completed_callback you presented here). All nodes in the callback must be the k closest nodes to the target.

So in my mind, it's just the completed_callback, the first step for put (announce/put) something to DHT. (The nodes for traversal logic is a internal logic, need call any callback)